src/main/kotlin/de/europace/gradle/docker/publish/DockerPublishPlugin.kt

```diff
@@ -69,7 +70,16 @@ class DockerPublishPlugin : Plugin<Project> {
     }
   }
 
-  private fun getOrganisation(extension: DockerPublishExtension): String {
-    return extension.organisation.getOrNull() ?: throw GradleException("organisation must be set")
+  private fun getOrganisation(project: Project, extension: DockerPublishExtension): Provider<String> {
+    return project.provider {
+      if (!extension.organisation.isPresent) {
+        throw GradleException("organisation must be set")
+      }
+      extension.organisation.get()
+    }
```

Comment:
what exactly is wrong with the other writing? I find the other one less verbose, and it does seem same overall?